contracts/staking/StakingPool.sol

```diff
@@ -591,17 +592,16 @@ contract StakingPool is
      */
     function _enforceUniqueRewardTokens(
         StakingPoolLib.Reward[] calldata rewardPools
-    ) private pure {
+    ) private {
         for (uint256 i = 0; i < rewardPools.length; i++) {
-            // Ensure no later entries contain the same tokens address
-            uint256 next = i + 1;
-            if (next < rewardPools.length) {
-                for (uint256 j = next; j < rewardPools.length; j++) {
-                    if (rewardPools[i].tokens == rewardPools[j].tokens) {
-                        revert("Rewards: tokens must be unique");
-                    }
-                }
-            }
+            // Ensure no prev entries contain the same tokens address
+            if (_tokensCounter[address(rewardPools[i].tokens)]) {
```

Comment:
suggestion: would it make sense to change the `revert` for a `require`?

_It may read easier as a require as we are not evaluating any complex expression i.e. it's a true or false_

Reply:
thanks, it will be easier to read.

Reply:
awaiting approval, thanks a lot.